integrations/cohere/tests/test_chat_generator.py

```diff
@@ -656,3 +657,254 @@ def test_live_run_multimodal(self):
         assert len(results["replies"]) == 1
         assert isinstance(results["replies"][0], ChatMessage)
         assert len(results["replies"][0].text) > 0
+
+
+class TestReasoningExtraction:
+    """Test the reasoning extraction functionality."""
+
+    def test_extract_reasoning_with_thinking_tags(self):
+        """Test extraction of reasoning from <thinking> tags."""
+        response_text = """<thinking>
+I need to calculate the area of a circle.
+The formula is π * r².
+Given radius is 5, so area = π * 25 = 78.54
+</thinking>
+
+The area of a circle with radius 5 is approximately 78.54 square units."""
```

Comment:
Hey @ChinmayBansal is it really the case that Cohere only returns reasoning content as part of the text response wrapped in these tags?